### ISSUE TITLE
Screen Reader: Many Pages: The text field is announced without a role

### ISSUE THREAD

**Issue by olenamihno** [Expensify/App#77569]
## Action Performed:
Prerequisites:
- the user is not signed in

1. Using iOS VoiceOver or Android TalkBack, open the Expensify app
2. Swipe to the 'Phone or email' text field
3. Note how the field is announced.

Other Occurrences:
1. Onboarding: all fields in the flow
2. On Inbox: the field 'Enter your boss's email'
3. On Status: the field 'Message'
4. On Status - Vacation Delegate
5. On Profile - Display Name
6. On Profile - Address
7. On Profile - Legal Name
8. On Security - Merge Accounts
9. On Security - Close Account
10. On Wallet - Add Bank Account
11. On Save the World - I know a teacher
12. On Filters
13. On Create Workspace

## Expected Result:
Each text field should be announced with a text field role.

## Actual Result:
The text field is announced without a role.

## Area issue was found in:
Sign in

## Failed WCAG checkpoints
WCAG 4.1.2 Name, Role, Value
https://www.w3.org/WAI/WCAG22/Understanding/name-role-value

## User impact:
Providing role, state, and value information on all user interface components enables compatibility with assistive technology, such as screen readers, screen magnifiers, and speech recognition software, used by people with disabilities.

## Suggested resolution:
Ensure each input field is implemented correctly, so that its role, name and value are announced.
https://reactnative.dev/docs/textinput

## Environment:
Samsung Galaxy S24 Ultra - Android 14, Partner - Native (No mobile browser)
Apple iPhone 11 - iOS 17.5.1, H0T Mobile - Native (No mobile browser)

## Screenshots/Videos

![Image](https://github.com/user-attachments/assets/b2db5f6e-66f1-4e14-9899-d74baa6ddd40)

https://github.com/user-attachments/assets/63b5b2fb-f812-4707-96f8-2c0143fc41e5

**Version Number:** Production
**Reproducible in staging?:** NA
**Reproducible in production?:** Yes
**Email or phone of affected tester (no customers):** No
**Logs:** https://stackoverflow.com/c/expensify/questions/4856  NA
**Notes/Photos/Videos:** Any additional supporting documentation  Yes
**Expensify/Expensify Issue URL:** https://platform.applause.com/company/12102/products/36628/community-issues/7004813
**Issue reported by:** NA
**Slack conversation:** NA
